UCT/ROCM: fix test_md_fork

edgargabriel opened the pull request:
## What
fix for the test_md_fork test of gtest for ROCm devices and memory domains.

yosefe commented on test/gtest/uct/test_md.cc:
```diff
@@ -1067,7 +1067,11 @@ UCS_TEST_SKIP_COND_P(test_md_fork, fork,
         memset(buf, 42, REG_SIZE);
         /* child touch the page */
         EXPECT_EQ(0, memcmp(page, buf, REG_SIZE));
+#if HAVE_ROCM
+        _exit(0);
+#else
         exit(0);
+#endif
```
why needed?

yosefe replied:
@edgargabriel can we avoid this in the code and solve in a better way?

edgargabriel replied:
@yosefe yes, I was told that the runtime fix have a solution for the fork cleanup problem implemented, and I am just waiting to be able to test that. If that works indeed, I would just keep the first part of the PR and remove the '_exit' from it

yosefe replied:
> @yosefe yes, I was told that the runtime folks have a fix for the fork cleanup problem implemented, and I am just waiting to be able to test that. If that works indeed, I would just keep the first part of the PR and remove the '_exit' from it

sounds good

yosefe commented:
/azp run UCX PR

edgargabriel commented:
@yosefe I've updated the pr to contain only the change to the rcache_params, and removed the change to gtest. Please have look when you have a chance.

yosefe reviewed (approved):
pls squash
